pkg/testing/tools/artifacts_api.go

```diff
@@ -0,0 +1,224 @@
+	Manifests struct {
+		LastUpdateTime         string `json:"last-update-time"`
+		SecondsSinceLastUpdate int    `json:"seconds-since-last-update"`
+	} `json:"manifests"`
+}
+
+type Package struct {
+	URL        string `json:"url"`
+	ShaURL     string `json:"sha_url"`
+	AscURL     string `json:"asc_url"`
+	Type       string `json:"type"`
+	Attributes struct {
+		ArtifactNoKpi string `json:"artifactNoKpi"`
+		Internal      string `json:"internal"`
+		ArtifactID    string `json:"artifact_id"`
+		Oss           string `json:"oss"`
+		Group         string `json:"group"`
+	} `json:"attributes"`
+}
+
+type Project struct {
+	Branch                       string             `json:"branch"`
+	CommitHash                   string             `json:"commit_hash"`
+	CommitURL                    string             `json:"commit_url"`
+	ExternalArtifactsManifestURL string             `json:"external_artifacts_manifest_url"`
+	BuildDurationSeconds         int                `json:"build_duration_seconds"`
+	Packages                     map[string]Package `json:"packages"`
+	Dependencies                 []any              `json:"dependencies"`
+}
+
+type BuildDetails struct {
+	Build struct {
+		Projects             map[string]Project `json:"projects"`
+		StartTime            string             `json:"start_time"`
+		ReleaseBranch        string             `json:"release_branch"`
+		Prefix               string             `json:"prefix"`
+		EndTime              string             `json:"end_time"`
+		ManifestVersion      string             `json:"manifest_version"`
+		Version              string             `json:"version"`
+		Branch               string             `json:"branch"`
+		BuildID              string             `json:"build_id"`
+		BuildDurationSeconds int                `json:"build_duration_seconds"`
+	} `json:"build"`
+	Manifests struct {
+		LastUpdateTime         string `json:"last-update-time"`
+		SecondsSinceLastUpdate int    `json:"seconds-since-last-update"`
+	} `json:"manifests"`
+}
+
+type SearchPackageResult struct {
+	Packages  map[string]Package `json:"packages"`
+	Manifests struct {
+		LastUpdateTime         string `json:"last-update-time"`
+		SecondsSinceLastUpdate int    `json:"seconds-since-last-update"`
+	} `json:"manifests"`
+}
+
+type httpDoer interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+type ArtifactAPIClientOpt func(aac *ArtifactAPIClient)
+
+func WithUrl(url string) ArtifactAPIClientOpt {
+	return func(aac *ArtifactAPIClient) { aac.url = url }
+}
+
+func WithHttpClient(client httpDoer) ArtifactAPIClientOpt {
+	return func(aac *ArtifactAPIClient) { aac.c = client }
+}
+
+// ArtifactAPIClient is a small (and incomplete) client for the Elastic artifact API.
+// More information about the API can be found at https://artifacts-api.elastic.co/v1
+// which will print a list of available operations
+type ArtifactAPIClient struct {
+	c   httpDoer
+	url string
+}
+
+// Creates a new Artifact API client
```

Comment:
Nit: It's more Golang [convention](https://go.dev/blog/godoc) but comments are supposed to be complete sentences, starting with the name of the element being described.

```suggestion
// NewArtifactAPIClient creates a new Artifact API client
```